ui/src/main/java/com/wireguard/android/Application.kt

```diff
@@ -94,6 +99,41 @@ class Application : android.app.Application() {
         return backend
     }
 
+    private fun registerForNetworkChanges() {
+        val cm = applicationContext.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+
+        val callback = object : ConnectivityManager.NetworkCallback() {
```

Comment:
should we be unregistering the callback in `onTerminate`?

Reply:
It should make no functional difference as the listener is tied to the application lifetime, but I can see the value of adding it anyway